discretize/MeshIO.py

```diff
@@ -118,51 +124,60 @@ def unpackdx(fid, nrows):
         return tensMsh
 
     @classmethod
-    def readUBC(TensorMesh, fileName, meshdim=None):
+    def readUBC(TensorMesh, fileName, meshdim=None, directory=''):
         """Wrapper to Read UBC GIF 2D  and 3D tensor mesh and generate same dimension TensorMesh.
 
-        :param string fileName: path to the UBC GIF mesh file
+        Input:
+        :param str fileName: path to the UBC GIF mesh file or just its name if directory is specified
+        :param str directory: directory where the UBC GIF file lives
         :param int meshdim: expected dimension of the mesh, if unknown the default argument is None
+
+        Output:
         :rtype: TensorMesh
         :return: The tensor mesh for the fileName.
         """
         # Check the expected mesh dimensions
+        fname = os.path.join(directory, fileName)
         if meshdim == None:
             # Read the file as line strings, remove lines with comment = !
-            msh = np.genfromtxt(fileName, delimiter='\n', dtype=np.str, comments='!', max_rows=1)
+            msh = np.genfromtxt(fname, delimiter='\n', dtype=np.str, comments='!', max_rows=1)
             # Fist line is the size of the model
             sizeM = np.array(msh.ravel()[0].split(), dtype=float)
             # Check if the mesh is a UBC 2D mesh
             if sizeM.shape[0] == 1:
-                Tnsmsh = TensorMesh._readUBC_2DMesh(fileName)
+                Tnsmsh = TensorMesh._readUBC_2DMesh(fname)
```

Comment:
to simplify this a bit (and reduce redundancy), why don't we just assign `meshdim = 2` here or `meshdim = 3` below, and then  check meshdim and call the appropriate `readUBC` method after that. 

Thoughts?

Reply:
I agree, this is a function I wrote in a previous PR but it was a redundant check between what the user expects and what the machine read.